### DISCUSSION TITLE
Observability for streaming models

### DISCUSSION THREAD

**Discussion by cmasantos** [quarkiverse/quarkus-langchain4j#1080] [Q&A]
Hi, I've notice that "normal" chat model produce a set of `gen_ai*` metrics. But I've been unable to use them with streaming models. Also when we use a combination of Mutiny StreamingResponseHandler I'm almost sure that it can't be done.

My question is, how can I add metrics to a streaming model even if need to instantiate them? I've found the `MetricsChatModelListener` class but I can't find a way to put it together with the streaming model interface.

**Comment by geoand** [comment]
Good idea to bring it up!

I opened #1081
